src/executor.rs

```diff
@@ -25,161 +30,90 @@ pub struct Task {
-                                            })
-                                        }
-                                    };
-                                    if let Err(e) = line_sender.send(r) {
-                                        debug!("error when sending stdout line: {}", e);
-                                        break;
+
+                            let child = match start_task(task, &mut command) {
+                                Err(e) => {
+                                    let response = CommandExecInfo::Error(
+                                        format!("failed to start task: {}", e)
+                                    );
+                                    match line_sender.send(response) {
+                                        Err(_) => break,
+                                        _ => continue,
                                     }
-                                    buf.clear();
-                                }
-                            }))
-                        } else {
-                            None
-                        };
-                        let mut buf = String::new();
-                        let mut buf_reader = BufReader::new(stderr);
-                        loop {
-                            if let Ok(()) = stop_receiver.try_recv() {
-                                debug!("stopping during execution");
-                                match child.kill() {
-                                    Ok(()) => debug!("command stopped"),
-                                    _ => debug!("command already stopped"),
-                                }
-                                return;
-                            }
-                            let r = match buf_reader.read_line(&mut buf) {
-                                Err(e) => CommandExecInfo::Error(e.to_string()),
-                                Ok(0) => {
-                                    // finished
-                                    break;
-                                }
-                                Ok(_) => {
-                                    // debug!("STDERR : {:?}", &buf);
-                                    CommandExecInfo::Line(CommandOutputLine {
-                                        origin: CommandStream::StdErr,
-                                        content: TLine::from_tty(buf.trim_end()),
-                                    })
                                 }
+                                Ok(child) => child,
                             };
-                            if let Err(e) = line_sender.send(r) {
-                                debug!("error when sending stderr line: {}", e);
-                                break;
-                            }
-                            buf.clear();
+
+                            current_task = Some(tokio::spawn(                                execute_task(child, with_stdout, line_sender.clone())
```

Comment:
SPAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACE ?!?

Reply:
Not sure what you are reffering to, could you be more specific?

Reply:
There's a formatting problem. But it doesn't matter much, I'll reformat if there's nothing more important.

Reply:
Never mind, I see it now... had to scroll more to the right. Strange that `rustfmt` left it that way...